modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -130,16 +130,17 @@ export default class TileLayer extends CompositeLayer {
       const isVisible = visible && tile.isVisible;
       // cache the rendered layer in the tile
       if (!tile.layer) {
-        tile.layer = renderSubLayers(
+        const layers = renderSubLayers(
           Object.assign({}, this.props, {
             id: `${this.id}-${tile.x}-${tile.y}-${tile.z}`,
             data: tile.data,
             visible: isVisible,
             tile
           })
         );
-      } else if (tile.layer.props.visible !== isVisible) {
-        tile.layer = tile.layer.clone({visible: isVisible});
+        tile.layer = flatten(layers, Boolean);
```

Comment:
Isn't strange to have a variable called `layer` holding an array? Should it be called `tile.layers`